src/System.Linq.Parallel/tests/ExchangeTests.cs

```diff
@@ -10,183 +11,180 @@ namespace Test
 {
     public class ExchangeTests
     {
-        [Fact]
-        public static void SimplePartitionMergeWhereScanTest()
+        private static readonly ParallelMergeOptions[] Options = new[] {
+            ParallelMergeOptions.AutoBuffered,
+            ParallelMergeOptions.Default,
+            ParallelMergeOptions.FullyBuffered,
+            ParallelMergeOptions.NotBuffered
+        };
+
+        // For each source, test with different degrees of parallelism to force PLINQ to partition.
+        public static IEnumerable<object[]> PartitioningData(object[] counts)
         {
-            for (int i = 1; i <= 128; i *= 2)
+            foreach (object[] results in Sources.Ranges(counts.Cast<int>(), x => new[] { 1, 2, 4 }))
             {
-                SimplePartitionMergeWhereScanTest1(1024 * 8, i, true);
-                SimplePartitionMergeWhereScanTest1(1024 * 8, i, false);
+                yield return results;
             }
         }
 
-        [Fact]
-        [OuterLoop]
-        public static void SimplePartitionMergeWhereScanTest_LongRunning()
+        // For each source, run with each buffering option.
+        public static IEnumerable<object[]> MergeData(object[] counts)
         {
-            for (int i = 1; i <= 128; i *= 2)
+            foreach (object[] results in Sources.Ranges(counts.Cast<int>(), x => Options))
             {
-                SimplePartitionMergeWhereScanTest1(1024 * 1024 * 2, i, true);
-                SimplePartitionMergeWhereScanTest1(1024 * 1024 * 2, i, false);
+                yield return results;
             }
         }
 
-        private static void SimplePartitionMergeWhereScanTest1(int dataSize, int partitions, bool pipeline)
+        // For each count, return an Enumerable source that fails (throws an exception) on that count,
+        // with each buffering option.
```

Comment:
That description is helpful, thanks.  Should the method itself be renamed then to something more descriptive, i.e. instead of ```FailingMergeData```, something like ```ThrowOnCount_AllMergeOptions_MemberData```?  I'm trying to come up with ways that make the tests self-explanatory (to folks besides you :smile:) without having to do lots of exploring and code reading.